staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -974,31 +1065,41 @@ func (c *cacheWatcher) add(event *watchCacheEvent, timer *time.Timer, budget *ti
 	budget.returnUnused(timeout - time.Since(startTime))
 }
 
+func (c *cacheWatcher) nextBookmarkTime(now time.Time) time.Time {
+	// For now we return 2s before deadline (and maybe +infinity is now already passed this time)
+	// but it gives us extensibility for the future.
```

Comment:
What if deadline -2s is before now?

This may lead to adding it to the bucket that has already been processed and we will never remove the watcher from there (and leak it effectively)?

Reply:
Adding to the first bucket to get a change to be triggered.